### PR TITLE
[FONT][FREETYPE] Porting Wine font engine Part.1

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose
For porting Wine font engine code. This will fix Anastasia font issue.
JIRA issue: [CORE-13269](https://jira.reactos.org/browse/CORE-13269)

## Proposed changes

- Improve EngWideCharToMultiByte and EngMultiByteToWideChar functions.
- Implement get_glyph_index and get_default_char_index functions etc.
- Implement get_face_name function from Wine as replacement of IntGetFontLocalizedName.
- Implement pick_charmap function from Wine.
- Rename and add some FONTGDI members.

### CODE REVIEW

**Comment by katahiromz** [comment]
BEFORE:
anastasia-before
AFTER:
anastasia-after

File: win32ss/gdi/ntgdi/freetype.c [hunk 1]
```diff
@@ -903,6 +903,7 @@ IntGdiLoadFontsFromMemory(PGDI_LOAD_FONT pLoadFont,
     /* set face */
     FontGDI->SharedFace = SharedFace;
     FontGDI->charset = ANSI_CHARSET;
+    FontGDI->codepage = 1252;
```
Comment (HBelusca) [inline win32ss/gdi/ntgdi/freetype.c]: Add a comment about what CP this number corresponds to? (windows-1252 | ANSI Latin 1; Western European (Windows).)

File: win32ss/gdi/ntgdi/freetype.c [hunk 2]
```diff
@@ -988,6 +1018,7 @@ IntGdiLoadFontsFromMemory(PGDI_LOAD_FONT pLoadFont,
     if (RtlEqualUnicodeString(&Entry->FaceName, &g_MarlettW, TRUE))
     {
         FontGDI->charset = SYMBOL_CHARSET;
+        FontGDI->codepage = IntCodePageFromCharSet(SYMBOL_CHARSET);
```
Comment (HBelusca) [inline win32ss/gdi/ntgdi/freetype.c]: Is SYMBOL_CHARSET also equal to FT_WinFNT_ID_SYMBOL ?

Comment (katahiromz) [reply win32ss/gdi/ntgdi/freetype.c]: Grep'ed.

```cpp
// <wingdi.h>
#define SYMBOL_CHARSET 2

// <ftwinfnt.h>
#define FT_WinFNT_ID_SYMBOL    2
```

File: win32ss/gdi/ntgdi/freetype.c (hunk 1 above)
Comment (katahiromz) [reply win32ss/gdi/ntgdi/freetype.c]: OK.

**Comment by katahiromz** [comment]
Time up.

**Comment by HBelusca** [comment]
Well I was planning to review that PR during the weekend....

**Pull request closed without merging**